src/frontend/workers/folderWatcher.worker.ts

```diff
@@ -63,10 +92,16 @@ export class FolderWatcherWorker {
           }
         })
         // .on('change', (path: string) => console.debug(`File ${path} has been changed`))
+        // TODO: on directory change: update location hierarchy list
         .on('unlink', (path: string) => ctx.postMessage({ type: 'remove', value: path }))
         .on('ready', () => {
           this.isReady = true;
           resolve(initialFiles);
+
+          // TODO: Clear memory: initialFiles no longer needed
+          // Update: tried it, didn't work as expected: list was emptied before sent back to main thread
+          // maybe send a message from main thread after initialization is finished instead?
+          // initialFiles.splice(0, initialFiles.length);
```

Comment:
What exactly is the problem?

Reply:
It always resolves to an empty array on the main thread if you try to clear the array in the web worker? I think that the Promise is only resolved after the last line of code. Regardless, where you call resolve does not matter as long it is still the same executing block. At least that is how I imagine it works with my wonky async know how.